pkg/diagnostics/workflow_monitoring.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnostics
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+
+	diagUtils "github.com/dapr/dapr/pkg/diagnostics/utils"
+)
+
+var executionTypeKey = tag.MustNewKey("execution_type")
+
+const (
+	StatusSuccess     = "success"
+	StatusFailed      = "failed"
+	StatusRecoverable = "recoverable"
+	CreateWorkflow    = "create_workflow"
+	GetWorkflow       = "get_workflow"
+	AddEvent          = "add_event"
+	PurgeWorkflow     = "purge_workflow"
+	Activity          = "activity"
+	Workflow          = "workflow"
+
+	WorkflowEvent = "event"
+	Timer         = "timer"
+
+	ComponentName = "dapr"
+)
+
+type workflowMetrics struct {
+	// workflowOperationCount records count of Successful/Failed requests to Create/Get/Purge Workflow and Add Events.
+	workflowOperationCount *stats.Int64Measure
+	// workflowOperationLatency records latency of response for workflow operation requests.
+	workflowOperationLatency *stats.Float64Measure
+	// workflowExecutionCount records count of Successful/Failed/Recoverable workflow/activity executions.
+	workflowExecutionCount *stats.Int64Measure
+	// workflowExecutionLatency records latency of workflow/activity executions.
+	workflowExecutionLatency *stats.Float64Measure
+	appID                    string
+	enabled                  bool
+	namespace                string
+}
+
+func newWorkflowMetrics() *workflowMetrics {
+	return &workflowMetrics{
+		workflowOperationCount: stats.Int64(
+			"runtime/workflow/operation/count",
+			"The number of successful/failed workflow operation requests.",
+			stats.UnitDimensionless),
+		workflowOperationLatency: stats.Float64(
+			"runtime/workflow/operation/latency",
+			"The latencies of responses for workflow operation requests.",
+			stats.UnitMilliseconds),
+		workflowExecutionCount: stats.Int64(
+			"runtime/workflow/execution/count",
+			"The number of successful/failed/recoverable workflow/activity executions.",
+			stats.UnitDimensionless),
+		workflowExecutionLatency: stats.Float64(
+			"runtime/workflow/execution/latency",
```

Comment:
Should it be `runtime/workflow/execution/latency` and `runtime/workflow/activity/execution/latency` - differentiation would be there - and cardinality can also be reduced by removing executionTypeKey.

Reply:
Similarly, for workflow/execution/count metric as well

Reply:
Sounds good to me

Reply:
Sounds good to me.